pkg/fanal/image/remote.go

```diff
@@ -67,26 +67,21 @@ func tryRemote(ctx context.Context, imageName string, ref name.Reference, option
 
 }
 
-func parsePlatform(ref name.Reference, p string) (*v1.Platform, error) {
+func parsePlatform(ref name.Reference, options []remote.Option, p string) (*v1.Platform, error) {
 	// OS wildcard, implicitly pick up the first os found in the image list.
 	// e.g. */amd64
 	if strings.HasPrefix(p, "*/") {
-		index, err := remote.Index(ref)
+		index, err := remote.Index(ref, options...)
 		if err != nil {
-			// Not a multi-arch image
-			if _, ok := err.(*remote.ErrSchema1); ok {
-				log.Logger.Debug("Ignored --platform as the image is not multi-arch")
-				return nil, nil
-			}
-			return nil, xerrors.Errorf("remote index error: %w", err)
+			return nil, err
```

Comment:
We should wrap the error.

```suggestion
			return nil, xerrors.Errorf("remote index error: %w", err)
```